package.json

```diff
@@ -16,7 +16,8 @@
     "precoverage": "npm install",
     "coverage": "node $HARMONY_OPTS ./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha",
     "precoveralls": "npm run coverage",
-    "coveralls": "coveralls < coverage/lcov.info"
+    "coveralls": "coveralls < coverage/lcov.info",
+    "lint": "./node_modules/.bin/eslint index.js lib/* bin/* test/*.js"
```

Comment:
Can we have Travis run this so the lint rules are actually enforced?

You may also need to `git cherry-pick` the necessary style changes from the async-await branch, as I suspect the lint currently fails as-is?

Reply:
> Can we have Travis run this so the lint rules are actually enforced?

I suppose… are you certain you want that?

I love lint tools, but I never liked contributing to projects that refuse PRs when a lint fails.  IMO the practice discourages contributions—especially from newbies to open source, or to a particular programming language.  (In other words, precisely the people who need positive, accepting experiences when they start out.)

I believe this is also doubly important for projects like Metalsmith—projects which are trying to bounce back after a prolonged drop-off in active development.  

(I want to see Metalsmith come back and attract a huge following!  I truly believe it has the right combination of architecture, flexibility, and utility to solve real-world problems.  The only thing it _doesn’t_ have at the moment, is any guarantee of longevity.  I’m hoping to help turn that around.  ☻)

If you disagree, I’ll add it to the Travis script.  But I’d rather have lint tools display warnings and leave it to humans to decide whether to merge or say “fix those lint items”.

Reply:
That's an interesting perspective. I've always treated lint warnings as something that always needs to be either fixed or explicitly ignored. If you just leave hundreds of unaddressed warnings lying around in your codebase, doesn't it become really easy to miss new warnings that pop up during future changes? And why is being required to fix lint errors any more discouraging for newbies than being required to fix failing tests?

Regardless of whether we want to enforce lint rules on PRs or not, the lint does need to be run by Travis so people can see the lint results without having to manually check out the branch run the lint themselves. If we don't want to enforce the rules, that particular Travis job can be marked with `allow_failures` in the config, but it _does_ still need to run.

Reply:
> I've always treated lint warnings as something that always needs to be either fixed or explicitly ignored.

Actually, I totally agree!

> And why is being required to fix lint errors any more discouraging for newbies than being required to fix failing tests?

It’s not, actually.  In fact, lint errors are way easier to fix.

----

## Flashback: Hobgoblins of Little Minds

I think you just helped me do a little therapeutic self-reflection.  I believe I was carrying around some frustration I had with past projects.  Specifically, projects that stuck to the lint rules so blindly, that they refused any attempted negotiation, and actually enforced rules that actually made certain bits of code harder to read.  

These projects were all written in Python.  I love the Python language, the Python community in general, and I largely agree with the officially recommended Python code style.  The style is known as “[PEP8](https://www.python.org/dev/peps/pep-0008/)”, and there’s a package with a command to check code against the style.

However, although the PEP8 document—_immediately after the introduction, in the memorably titled section, “A Foolish Consistency is the Hobgoblin of Little Minds”_—the document explicitly says that it’s okay to deviate from the style guide wherever it makes sense and improves code readability.

Unfortunately, for all the goodness in the Python community, lots and lots of projects blindly enforce the PEP8 style with _absolutely **no** exceptions_.  

I was repeatedly frustrated out of contributing to projects because of it.

----

## …Back to Here and Now

(Ahh… I think I feel lighter, now. ☻)

But this situation is different.  The language, the tools, and the community is different.  And ESLint is customizable—right down to individual lines of code, if necessary.  

…What an odd thing for me to have been carrying around all this time!  Thanks for helping me uncover it.

----

I’ll add the lint to Travis shortly. 

*high five*

Reply:
👍

Reply:
I have [lint running for Node v10](https://travis-ci.org/Zearin/metalsmith/jobs/386251781#L838), but [not](https://travis-ci.org/Zearin/metalsmith/jobs/386251780#L841) [on](https://travis-ci.org/Zearin/metalsmith/jobs/386251779#L841) [other](https://travis-ci.org/Zearin/metalsmith/jobs/386251777#L746) [versions](https://travis-ci.org/Zearin/metalsmith/jobs/386251775#L576).